### PR TITLE
dmctl: fix dmctl command to improve its usability

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Previous dmctl has to put `--master-addr` in the front. What's more, we hack many usage which is not standard usage of cobra commands.

### What is changed and how it works?
1. Rewrite the cobra command deployment to make dmctl more easy to use. Now `--master-addr` can be put any where.
2. Mark `--decrypt/encrypt` as hidden. Add `encrypt/decrypt` commands.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### CODE REVIEW

File: dm/ctl/ctl.go [hunk 1]
```diff
@@ -108,48 +107,155 @@ func Init(cfg *common.Config) error {
...
+func Start(args []string) (err error) {
+	commandMasterFlags.Reset()
+	rootCmd := NewRootCmd()
+	rootCmd.SetArgs(args)
+	return rootCmd.Execute()
+}
+
+func loop() error {
+	l, err := readline.NewEx(&readline.Config{
+		Prompt:          "\033[31m»\033[0m ",
```
Comment (Ehco1996) [inline dm/ctl/ctl.go:120]: magic...

File: tests/handle_error_3/lib.sh [hunk 2]
```diff
@@ -45,31 +45,31 @@ function clean_table() {
...
-	pos=$($PWD/bin/dmctl.test DEVEL -master-addr=$1 \
+	pos=$($PWD/bin/dmctl.test DEVEL --master-addr=$1 \
 		query-status test -s $2 |
 		grep Location |
 		gawk 'match($0,/startLocation: \[position: \((.*), (.*)\), gtid-set: (.*)\], endLocation: \[position: \((.*), (.*)\), gtid-set: (.*)\]/,a) {print a[2]}')
 	echo ${pos}
 }
 
 function get_start_name() {
-	pos=$($PWD/bin/dmctl.test DEVEL -master-addr=$1 \
```
Comment (lance6716) [inline tests/handle_error_3/lib.sh:72]: I see our stuff using one `-` sometimes, so we might keep compatibility

File: tests/dmctl_basic/check_list/dmctl.sh [hunk 3]
```diff
@@ -15,9 +15,11 @@ function dmctl_unwrap_schema() {
 function dmctl_wrong_addrs() {
 	run_dm_ctl $WORK_DIR "https://127.0.0.1:1234,127.0.0.2:1234" \
 		"list-member" \
-		"can't connect to https:\/\/127.0.0.1:1234,127.0.0.2:1234" 1
+		"can't connect to https:\/\/127.0.0.1:1234,127.0.0.2:1234" 2
```
Comment (lance6716) [inline tests/dmctl_basic/check_list/dmctl.sh:18]: now print error message twice 🤔

**Review by lance6716 — COMMENTED** [review]
will review later

File: tests/handle_error_3/lib.sh (hunk 2 above)
Comment (lichunzhu) [reply tests/handle_error_3/lib.sh:72]: addressed in https://github.com/pingcap/dm/pull/1750/commits/09244541025887ec0ec3e8777a1f67c5cec5b411

**Review by GMHDBJD — APPROVED** [review]
LGTM. Remember to update docs after merge.

**Review by lance6716 — APPROVED** [review]

File: dm/ctl/ctl.go [hunk 4]
```diff
@@ -108,48 +108,162 @@ func Init(cfg *common.Config) error {
...
+
+		err = cfg.Validate()
+		if err != nil {
+			return err
+		}
+
+		return Init(cfg)
+	}
+	common.DefineConfigFlagSet(rootCmd.PersistentFlags())
+	rootCmd.SetArgs(args)
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply dm/ctl/ctl.go:225]: mainly copied from pd-ctl. https://github.com/tikv/pd/blob/master/tools/pd-ctl/pdctl/ctl.go

**Pull request merged**